rotkehlchen/exchanges/manager.py

```diff
@@ -82,58 +88,88 @@ def edit_exchange(
         """Edits both the exchange object and the database entry
 
         Returns True if an entry was found and edited and false otherwise
-
-        May raise:
-        - InputError if there is an error updating the DB
         """
         exchangeobj = self.get_exchange(name=name, location=location)
         if not exchangeobj:
             return False, f'Could not find {str(location)} exchange {name} for editing'
 
-        # First edit the database entries. This may raise InputError
-        with self.database.user_write() as cursor:
-            self.database.edit_exchange(
-                cursor,
-                name=name,
-                location=location,
-                new_name=new_name,
-                api_key=api_key,
-                api_secret=api_secret,
-                passphrase=passphrase,
-                kraken_account_type=kraken_account_type,
-                binance_selected_trade_pairs=binance_selected_trade_pairs,
-                ftx_subaccount=ftx_subaccount,
-            )
+        # First validate exchange credentials
+        edited = exchangeobj.edit_exchange_credentials(ExchangeAuthCredentials(
+            api_key=api_key,
+            api_secret=api_secret,
+            passphrase=passphrase,
+            ftx_subaccount=ftx_subaccount,
+        ))
+        if edited is True:
+            try:
+                credentials_are_valid, msg = exchangeobj.validate_api_key()
+            except Exception as e:  # pylint: disable=broad-except
+                msg = str(e)
+                credentials_are_valid = False
+
+            if credentials_are_valid is False:
+                exchangeobj.reset_to_db_credentials()
+                return False, f'New credentials are invalid. {msg}'
+
+        # Then edit extra properties
+        success, msg = exchangeobj.edit_exchange_extras({
+            KRAKEN_ACCOUNT_TYPE_KEY: kraken_account_type,
+            BINANCE_MARKETS_KEY: binance_selected_trade_pairs,
+        })
```

Comment:
this shouldn't execute for all the exchanges, only for those that require modification of the extra data.

Also this is the old design too but if feels broken. Now is executed always because you don't know what exchanges should read from this properties (in this case kraken, binance and binance US) but it breaks the abstraction somehow and not the manager needs to know about the exchanges.

I would not touch anything and leave it as it is now but I don't like the design here